src/stream/src/executor/managed_state/aggregation/extreme.rs

```diff
@@ -154,13 +157,14 @@ impl<S: StateStore> GenericExtremeState<S> {
         }
     }
 
-    fn state_row_to_cache_entry(&self, state_row: &Row) -> (OrderedRow, Datum) {
-        let cache_key = OrderedRow::new(
+    fn state_row_to_cache_entry(&self, state_row: &Row) -> StreamExecutorResult<(Vec<u8>, Datum)> {
+        let ordered_row = OrderedRow::new(
             state_row.by_indices(&self.state_table_order_col_indices),
             &self.state_table_order_types,
         );
+        let cache_key = ordered_row.serialize()?;
```

Comment:
How about directly serializing here?

Reply:
Sure can. I just wanted to keep the code concise.